### PR TITLE
Listing any name encoded in the certificates's subject alternative names field

### PR DESCRIPTION
**Pull request by labeneator**
Changes to allow any subject alt  name to be decoded and listed. My use case calls for listing URIs listed in certificates.

PS: Looks like my vim is stripping away spaces on empty lines. Do you need me to resubmit the PR w/out the cleanup?

Cheers,
Lmwangi

### CODE REVIEW

**Comment by hiviah** [comment]
Yes, it would be much cleaner without the whitespace changes.

Also a small fix is necessary: `for component_type, name_list in san.values.items()` in x509_parse.py needs to be indented right (to be under `if tbs.subjAltNameExt:` condition), otherwise it will fail when a certificate has no Subject Alternative Name extension.

BTW did you try this with a certificate that uses IP Address in SAN list? I don't have one handy right now. In the RFC 5280 the IP address is encoded into octet string which IIRC means that the address is byte-packed, i.e. each byte represents one field of IPv4 address.

File: pyx509/x509_parse.py [hunk 1]
```diff
@@ -130,24 +130,25 @@ def x509_parse(derData):
...
+
 	#if tbs.netscapeCertTypeExt: #...partially implemented
-	
+
 	if tbs.subjAltNameExt:
 		print "\tSubject Alternative Name: is_critical:", tbs.subjAltNameExt.is_critical
 		san = tbs.subjAltNameExt.value
-		print "\t\tDNS names:", ",".join(san.names) #only DNS names supported now
-		
+        for component_type, name_list in san.values.items():
```
Comment (labeneator) [inline pyx509/x509_parse.py]: Sigh... That was a mistake :(

**Comment by labeneator** [comment]
Hi,
 Yes, the address is byte packed. I verified this by inspecting a certificate that I just generated:

```
Out[2]: defaultdict(<type 'list'>, {'iPAddress': [u'\x01\x02\x03\x04'], 'dNSName': [u'feel-good.org'], 'uniformResourceIdentifier': [u'authority://host/path']})
```
SubjectAltNameExt should probably unpack it into the expected quad octet representation. That would be more pythonic. Right?

**Comment by labeneator** [comment]
Closing in favour of a new PR for the same issue

**Comment by labeneator** [comment]
Closed in favour of #7

**Pull request closed without merging**